Intermittent test failure: `TestNotifier/MultipleSubscribersStress`

Not super common, but I've seen it a couple times so far.

```
--- FAIL: TestNotifier (0.00s)
    --- FAIL: TestNotifier/MultipleSubscribersStress (0.25s)
        logger.go:[22](https://github.com/riverqueue/river/actions/runs/8257741142/job/22588819288?pr=258#step:8:23)5: time=20[24](https://github.com/riverqueue/river/actions/runs/8257741142/job/22588819288?pr=258#step:8:25)-03-13T00:59:07.700Z level=INFO msg="Notifier: Listener connecting"
        logger.go:225: time=2024-03-13T00:59:07.700Z level=INFO msg="Notifier: Run loop started"
        logger.go:2[25](https://github.com/riverqueue/river/actions/runs/8257741142/job/22588819288?pr=258#step:8:26): time=2024-03-13T00:59:07.700Z level=INFO msg="Notifier: Notifier healthy"
        notifier_test.go:425: Sending notification on "test_topic1": msg0
        logger.go:225: time=2024-03-13T00:59:07.701Z level=INFO msg="Notifier: Listening on topic" topic=test_topic1
        notifier_test.go:425: Sending notification on "test_topic1": msg1
        logger.go:225: time=2024-03-13T00:59:07.892Z level=INFO msg="Notifier: Unlistening on topic" topic=test_topic1
        logger.go:225: time=2024-03-13T00:59:07.892Z level=INFO msg="Notifier: Listener closing"
        logger.go:225: time=2024-03-13T00:59:07.893Z level=INFO msg="Notifier: Run loop stopped"
        notifier_test.go:462: Channel  0 contains   2 message(s)
        notifier_test.go:462: Channel  1 contains   2 message(s)
        notifier_test.go:462: Channel  2 contains   2 message(s)
        notifier_test.go:462: Channel  3 contains   2 message(s)
        notifier_test.go:462: Channel  4 contains   2 message(s)
        notifier_test.go:462: Channel  5 contains   2 message(s)
        notifier_test.go:462: Channel  6 contains   2 message(s)
        notifier_test.go:462: Channel  7 contains   0 message(s)
        notifier_test.go:467: 
            	Error Trace:	/home/runner/work/river/river/internal/notifier/notifier_test.go:467
            	Error:      	Should NOT be empty, but was 0xc000[29](https://github.com/riverqueue/river/actions/runs/8257741142/job/22588819288?pr=258#step:8:30)11[40](https://github.com/riverqueue/river/actions/runs/8257741142/job/22588819288?pr=258#step:8:41)
            	Test:       	TestNotifier/MultipleSubscribersStress
FAIL
```

https://github.com/riverqueue/river/actions/runs/8257741142/job/22588819288

--- a/internal/notifier/notifier_test.go
+++ b/internal/notifier/notifier_test.go
@@ -410,13 +410,13 @@ func TestNotifier(t *testing.T) {
 			notifyChans[i] = make(chan TopicAndPayload, 1000)
 		}
 
-		var wg sync.WaitGroup
-
 		// Start a goroutine to send messages constantly.
-		shutdown := make(chan struct{})
-		wg.Add(1)
+		var (
+			sendNotificationsDone     = make(chan struct{})
+			sendNotificationsShutdown = make(chan struct{})
+		)
 		go func() {
-			defer wg.Done()
+			defer close(sendNotificationsDone)
 
 			ticker := time.NewTicker(10 * time.Millisecond)
 			defer ticker.Stop()
@@ -427,7 +427,7 @@ func TestNotifier(t *testing.T) {
 				select {
 				case <-ctx.Done():
 					return
-				case <-shutdown:
+				case <-sendNotificationsShutdown:
 					return
 				case <-ticker.C:
 					// loop again
@@ -435,6 +435,7 @@ func TestNotifier(t *testing.T) {
 			}
 		}()
 
+		var wg sync.WaitGroup
 		wg.Add(len(notifyChans))
 		for i := range notifyChans {
 			notifyChan := notifyChans[i]
@@ -454,9 +455,9 @@ func TestNotifier(t *testing.T) {
 			}()
 		}
 
-		close(shutdown)
-		wg.Wait()
-		notifier.Stop()
+		wg.Wait()                        // wait for subscribe loops to finish all their work
+		close(sendNotificationsShutdown) // stop sending notifications
+		<-sendNotificationsDone          // wait for notifications goroutine to finish
 
 		for i := range notifyChans {
 			t.Logf("Channel %2d contains %3d message(s)", i, len(notifyChans[i]))

